src/Stan/Report/Html.hs

```diff
@@ -41,334 +40,318 @@ import Stan.Severity (Severity (..), severityDescription)
+stanExtensions from exts = divClass "col-6" $ do
+    button ! (A.class_ "collapsible") $ toHtml $ "Extensions from " <> from
+    ol ! (A.class_ "content") $ sequence_ $ map (li . toHtml) exts
+
+inspectionLink :: Id Inspection -> Html
+inspectionLink ins = a ! A.class_ "ins-link" ! A.href (fromString $ "#" <> toString insId) $ toHtml insId
   where
     insId :: Text
     insId = unId ins
 
-stanObservation o@Observation{..} = divIdClass (unId observationId) "observation row"
-    ( general
-    # pre_ (unlines $ prettyObservationSource False o)
-    # solutionsDiv inspection
-    )
+stanObservation :: Observation -> Html
+stanObservation o@Observation{..} = divIdClass (toString $ unId observationId) "observation row" $ do
+    general
+    pre $ toHtml (unlines $ prettyObservationSource False o)
+    solutionsDiv inspection
   where
-    general = divClass "observation-general" $ tableWithShadow ""
-        ( tableR "ID"            (unId observationId)
-        # tableR "Severity"      (severityFromIns inspection)
-        # tableR "Description"   (inspectionDescription inspection)
-        # tableR "Inspection ID" (inspectionLink observationInspectionId)
-        # tableR "Category"      (categories "inline" $ inspectionCategory inspection)
-        # tableR "File"          observationFile
-        )
-
-    tableR name val = tr_
-        ( td_A (A.class_ "info-name very-light-bg") name
-        # td_A (A.class_ "info-data") val
-        )
+    general = divClass "observation-general" $ tableWithShadow "" $ do
+        tableR "ID"            (unId observationId)
+        tableR "Severity"      (severityFromIns inspection)
+        tableR "Description"   (inspectionDescription inspection)
+        tableR "Inspection ID" (inspectionLink observationInspectionId)
+        tableR "Category"      (categories "inline" $ inspectionCategory inspection)
+        tableR "File"          observationFile
+
+    tableR :: ToMarkup a => Text -> a -> Html
+    tableR name val = tr $ do
+        td ! (A.class_ "info-name very-light-bg") $ toHtml name
+        td ! (A.class_ "info-data") $ toHtml val
 
     inspection :: Inspection
     inspection = getInspectionById observationInspectionId
 
+severityFromIns :: Inspection -> Html
 severityFromIns ins = severity $ show @Text $ inspectionSeverity ins
 
-severity severityTxt = span_A (A.class_ "severity")
-    ( span_A (A.class_ $ toText "severity" <> severityTxt) ""
-    # span_A (A.class_ "severityText") severityTxt
-    )
+severity :: Text -> Html
+severity severityTxt = span ! (A.class_ "severity") $ do
+    span ! (A.class_ $ fromString $ "severity" <> toString severityTxt) $ toHtml @Text ""
+    span ! (A.class_ "severityText") $ toHtml severityTxt
 
-categories cl cats = ul_A (A.class_ $ "cats " <> cl) $
-    map (li_A (A.class_ "cat") . unCategory) $ toList cats
+categories :: Text -> NonEmpty Category -> Html
+categories cl cats = ul ! (A.class_ $ fromString $ toString $ "cats " <> cl) $ sequence_ $
```

Comment:
Yeah, this looks like a pattern in many places... Maybe we can introduce a function

```haskell
fromText :: IsString s => Text -> s
```